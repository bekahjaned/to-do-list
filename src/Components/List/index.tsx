import React from "react";

import Task from "../Task/index";
import { ListWrapper } from "../../Elements/ListWrapper/index";

type Props = {
  tasks: string[];
};

class List extends React.Component<Props> {
  deleteTask = (index: number) => {
    let tasksArr: string[] = this.props.tasks;
    tasksArr.splice(index, 1);
    this.setState({ tasks: tasksArr });
  };

  render() {
    let tasks = this.props.tasks.map((val: string, key: number) => {
      return (
        <Task key={key} text={val} deleteMethod={() => this.deleteTask(key)} />
      );
    });

    return <ListWrapper>{tasks}</ListWrapper>;
  }
}

export default List;
