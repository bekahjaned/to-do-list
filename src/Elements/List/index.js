import styled from "styled-components";

export const List = styled.div`
  height: 385px;
  margin: 0 auto;
`;
