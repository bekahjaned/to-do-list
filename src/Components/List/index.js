import React from "react";

import Task from "../Task/";
import { ListWrapper } from "../../Elements/ListWrapper/";

class List extends React.Component {
  deleteTask(index) {
    let tasksArr = this.props.tasks;
    tasksArr.splice(index, 1);
    this.setState({ tasks: tasksArr });
  }

  render() {
    let tasks = this.props.tasks.map((val, key) => {
      return (
        <Task key={key} text={val} deleteMethod={() => this.deleteTask(key)} />
      );
    });

    return <ListWrapper>{tasks}</ListWrapper>;
  }
}

export default List;
