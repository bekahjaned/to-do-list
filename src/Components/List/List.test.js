import React from "react";
import renderer from "react-test-renderer";

import List from "./index";

it("render a Task", () => {
  // given
  const tasks = ["wash dishes", "do laundry"];

  // when
  const tree = renderer.create(<List tasks={tasks} />).toJSON();

  // then
  expect(tree).toMatchSnapshot();
});
