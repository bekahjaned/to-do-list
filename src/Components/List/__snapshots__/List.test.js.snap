// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a Task 1`] = `
<div
  className="sc-AxiKw bNTdIo"
>
  <div
    className="sc-AxirZ glbUEU"
  >
    <p>
      wash dishes
    </p>
    <button
      className="sc-AxjAm kuVXIB"
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <div
    className="sc-AxirZ glbUEU"
  >
    <p>
      do laundry
    </p>
    <button
      className="sc-AxjAm kuVXIB"
      onClick={[Function]}
    >
      x
    </button>
  </div>
</div>
`;
